cmd/crank/beta/top/top.go

```diff
@@ -0,0 +1,297 @@
+		logger.Debug("Printed pods summary")
+		fmt.Println()
+	}
+
+	if err := printPodsTable(k.Stdout, crossplanePods); err != nil {
+		return errors.Wrap(err, errPrintingPodsTable)
+	}
+	logger.Debug("Printed pods as table")
+	return nil
+}
+
+func printPodsTable(w io.Writer, crossplanePods []topMetrics) error {
+	tw := printers.GetNewTabWriter(w)
+	// Building header
+	headers := defaultPrinterRow{
+		podType:   "TYPE",
+		namespace: "NAMESPACE",
+		name:      "NAME",
+		cpu:       "CPU(milliCPU)",
+		memory:    "MEMORY(Mi)",
+	}
+	_, err := fmt.Fprintln(tw, headers.String())
+	if err != nil {
+		return errors.Wrap(err, errWriteHeader)
+	}
+
+	// Building rows for each pod
+	for _, pod := range crossplanePods {
+		row := defaultPrinterRow{
+			podType:   pod.PodType,
+			namespace: pod.PodNamespace,
+			name:      pod.PodName,
+			cpu:       fmt.Sprintf("%d", pod.CPUUsage),
+			memory:    pod.MemoryUsage,
+		}
+		_, err := fmt.Fprintln(tw, row.String())
+		if err != nil {
+			return errors.Wrap(err, errWriteRow)
+		}
+	}
+
+	return tw.Flush()
+}
+
+func printPodsSummary(w io.Writer, pods []topMetrics) {
+	categoryCounts := make(map[string]int)
+	totalCPUUsage := int64(0)
+	totalMemoryUsage := 0
+
+	for _, pod := range pods {
+		// Increment the count for this pod's category
+		categoryCounts[pod.PodType]++
+
+		// Aggregate CPU and Memory usage
+		totalCPUUsage += pod.CPUUsage
+		totalMemoryUsage += parseMemoryUsage(pod.MemoryUsage)
+	}
+
+	// Print summary directly to the provided writer
+	fmt.Fprintf(w, "Nr of Crossplane pods: %d\n", len(pods))
+	for category, count := range categoryCounts {
+		fmt.Fprintf(w, "%s: %d\n", capitalizeFirst(category), count)
+	}
+	fmt.Fprintf(w, "Memory(Mi): %dMi\n", totalMemoryUsage)
+	fmt.Fprintf(w, "CPU(milliCPU): %d\n", totalCPUUsage)
+}
+
+func getCrossplanePods(pods []v1.Pod) []topMetrics {
+	metricsList := make([]topMetrics, 0)
+	for _, pod := range pods {
+		labels := pod.GetLabels()
+
+		var podType string
+		isCrossplanePod := false
+		for labelKey, labelValue := range labels {
+			switch {
+			case strings.HasPrefix(labelKey, "pkg.crossplane.io/"):
+				podType = strings.SplitN(labelKey, "/", 2)[1]
+				if podType != "revision" {
+					isCrossplanePod = true
+				}
+			case labelKey == "app.kubernetes.io/part-of" && labelValue == "crossplane":
+				podType = "crossplane"
+				isCrossplanePod = true
+			}
+			if isCrossplanePod {
+				break
+			}
+		}
+
+		if isCrossplanePod {
+			metricsList = append(metricsList, topMetrics{
+				PodType:      podType,
+				PodName:      pod.Name,
+				PodNamespace: pod.Namespace,
+				CPUUsage:     0,
+				MemoryUsage:  "",
+			})
+		}
+	}
+	return metricsList
+}
+
+func capitalizeFirst(s string) string {
+	if s == "" {
+		return ""
+	}
+	return strings.ToUpper(s[:1]) + s[1:]
+}
+func parseMemoryUsage(memoryUsage string) int {
+	// Remove the 'Mi' suffix and parse the remaining number
+	trimmed := strings.TrimSuffix(memoryUsage, "Mi")
```

Comment:
you could use [resource.ParseQuantity(...)](https://github.com/kubernetes/apimachinery/blob/f14778da5523847e4c07346e3161a4b4f6c9186e/pkg/api/resource/quantity.go#L275) here directly, that would make it a little bit less brittle as it would handle other suffixes too.